gridstatus/miso.py

```diff
@@ -261,6 +261,74 @@ def _get_solar_and_wind_forecast_data(self, date, fuel, verbose=False):
 
         return df[self.solar_and_wind_forecast_cols]
 
+    @support_date_range(frequency="W-MON")
+    def get_lmp_weekly(self, date, end=None, verbose=False):
+        """Retrieves weekly lmp data that includes price corrections to the real time
+        data.
+
+        Data from: https://www.misoenergy.org/markets-and-operations/real-time--market-data/market-reports/#nt=%2FMarketReportType%3AHistorical%20LMP%2FMarketReportName%3AWeekly%20Real-Time%205-Min%20LMP%20(zip)&t=10&p=0&s=MarketReportPublished&sd=desc
+        """
+        if date in ["today", "latest"]:
+            raise NotSupported("Only historical data is available for weekly LMPs")
+
+        if not date.weekday() == 0:
+            log("Weekly LMP data is only available for Mondays", verbose)
+            log("Changing date to the previous Monday", verbose)
+            date -= pd.DateOffset(days=date.weekday())
+
+        # The data file contains data starting two weeks before the date and
+        # ending one week before the date. To get data that covers the date, we
+        # need to add two weeks to the date
+        date += pd.DateOffset(weeks=2)
+
+        download_url = f"https://docs.misoenergy.org/marketreports/{date.strftime('%Y%m%d')}_5MIN_LMP.zip"
+
+        df = pd.read_csv(download_url, compression="zip", skiprows=4, skipfooter=1)
+
+        return self._handle_lmp_weekly(df, verbose)
+
+    def _handle_lmp_weekly(self, df, verbose=False):
+        df["Interval Start"] = pd.to_datetime(df["MKTHOUR_EST"]).dt.tz_localize(
+            self.default_timezone,
+        )
+        df["Interval End"] = df["Interval Start"] + pd.Timedelta(minutes=5)
+
+        df = df.rename(
+            columns={
+                "CON_LMP": "Congestion",
+                "LOSS_LMP": "Loss",
+                "PNODENAME": "Location",
+            },
+        )
+        node_to_type = self._get_node_to_type_mapping(verbose)
+
+        df = df.merge(
+            node_to_type,
+            left_on="Location",
+            right_on="Node",
+            how="left",
+        )
+
+        df["Energy"] = df["LMP"] - df["Loss"] - df["Congestion"]
+        df["Market"] = Markets.REAL_TIME_5_MIN.value
```

Comment:
Should we add a new `REAL_TIME_5_MIN_WEEKLY` value to differentiate?

Reply:
yes, we should. 

in general I wish every lmp dataset was it's own function like we do for other datasets.

Reply:
Updated. Any further changes before this can be merged?